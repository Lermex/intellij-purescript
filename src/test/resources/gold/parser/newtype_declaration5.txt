Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('User7')
            PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('User7')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(ObjectType)
            PsiElement({)('{')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('name')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(TypeConstructor)
                    PSASTWrapperElement(Qualified)
                      PSASTWrapperElement(TypeConstructor)
                        PsiElement(proper name)('String')
              PsiElement(,)(',')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('isEnabled')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(TypeConstructor)
                    PSASTWrapperElement(Qualified)
                      PSASTWrapperElement(TypeConstructor)
                        PsiElement(proper name)('Boolean')
                        PsiElement(whitespace)(' ')
            PsiElement(})('}')