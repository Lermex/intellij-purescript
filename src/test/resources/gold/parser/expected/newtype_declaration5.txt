Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSNewTypeDeclarationImpl(NEW_TYPE_DECLARATION)
        PsiElement(PSTokenType.newtype)('newtype')
        PSDeclarationTypeImpl(DECLARATION_TYPE)
          PsiElement(PSTokenType.PROPER_NAME)('User7')
        PSIdListImpl(ID_LIST)
          PSDeclarationIdImpl(DECLARATION_ID)
            PsiElement(PSTokenType.ID)('a')
        PsiElement(PSTokenType.=)('=')
        PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
          PSTypeListImpl(TYPE_LIST)
            PSDeclarationTypeImpl(DECLARATION_TYPE)
              PsiElement(PSTokenType.PROPER_NAME)('User7')
          PSDataDeclarationBodyImpl(DATA_DECLARATION_BODY)
            PsiElement(PSTokenType.{)('{')
            PSDataDeclarationTypeDefinitionListImpl(DATA_DECLARATION_TYPE_DEFINITION_LIST)
              PSDataDeclarationTypeDefinitionImpl(DATA_DECLARATION_TYPE_DEFINITION)
                PsiElement(PSTokenType.ID)('name')
                PsiElement(PSTokenType.::)('::')
                PSParenthesisedTypeThenIdListImpl(PARENTHESISED_TYPE_THEN_ID_LIST)
                  PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                    PSDeclarationTypeImpl(DECLARATION_TYPE)
                      PsiElement(PSTokenType.PROPER_NAME)('String')
              PsiElement(PSTokenType.,)(',')
              PSDataDeclarationTypeDefinitionListImpl(DATA_DECLARATION_TYPE_DEFINITION_LIST)
                PSDataDeclarationTypeDefinitionImpl(DATA_DECLARATION_TYPE_DEFINITION)
                  PsiElement(PSTokenType.ID)('isEnabled')
                  PsiElement(PSTokenType.::)('::')
                  PSParenthesisedTypeThenIdListImpl(PARENTHESISED_TYPE_THEN_ID_LIST)
                    PSParenthesisedTypeListImpl(PARENTHESISED_TYPE_LIST)
                      PSDeclarationTypeImpl(DECLARATION_TYPE)
                        PsiElement(PSTokenType.PROPER_NAME)('Boolean')
            PsiElement(PSTokenType.})('}')